checks/evaluation/permissions/permissions.go

```diff
@@ -99,77 +138,215 @@ func applyScorePolicy(results *checker.TokenPermissionsData, c *checker.CheckReq
+			//nolint:gocritic
 			if r.LocationType == nil {
-				return checker.InconclusiveResultScore,
-					sce.WithMessage(sce.ErrScorecardInternal, "locationType is nil")
+				locationType = PermissionLocationNil
+			} else if *r.LocationType == checker.PermissionLocationTop {
+				locationType = PermissionLocationTop
+			} else {
+				locationType = PermissionLocationNotDeclared
 			}
+			permissionType := permTypeToEnum(r.Name)
+			f = f.WithValues(map[string]int{
+				"PermissionLevel": int(PermissionLevelUndeclared),
+				"LocationType":    int(locationType),
+				"PermissionType":  int(permissionType),
+			})
+		case checker.PermissionLevelWrite:
+			var locationType PermissionLocationType
+			switch *r.LocationType {
+			case checker.PermissionLocationTop:
+				locationType = PermissionLocationTop
+			case checker.PermissionLocationJob:
+				locationType = PermissionLocationJob
+			default:
+				locationType = PermissionLocationNotDeclared
+			}
+			permissionType := permTypeToEnum(r.Name)
+			f = f.WithValues(map[string]int{
+				"PermissionLevel": int(PermissionLevelWrite),
+				"LocationType":    int(locationType),
+				"PermissionType":  int(permissionType),
+			})
+			f = f.WithOutcome(finding.OutcomeNegative)
+		}
+		findings = append(findings, *f)
+	}
+	return findings, nil
+}
 
-			// We warn only for top-level.
-			if *r.LocationType == checker.PermissionLocationTop {
-				warnWithRemediation(dl, msg, remediationMetadata, loc, negativeProbeResults)
-			} else {
-				dl.Debug(msg)
+func permTypeToEnum(tokenName *string) PermissionType {
+	if tokenName == nil {
+		return PermissionTypeNone
+	}
+	switch *tokenName {
+	//nolint:goconst
+	case "all":
+		return PermissionTypeAll
+	case "statuses":
+		return PermissionTypeStatuses
+	case "checks":
+		return PermissionTypeChecks
+	case "security-events":
+		return PermissionTypeSecurityEvents
+	case "deployments":
+		return PermissionTypeDeployments
+	case "contents":
+		return PermissionTypePackages
+	case "actions":
+		return PermissionTypeActions
+	default:
+		return PermissionTypeNone
+	}
+}
+
+func permTypeToName(permType int) *string {
+	var permName string
+	switch PermissionType(permType) {
+	case PermissionTypeAll:
+		permName = "all"
+	case PermissionTypeStatuses:
+		permName = "statuses"
+	case PermissionTypeChecks:
+		permName = "checks"
+	case PermissionTypeSecurityEvents:
+		permName = "security-events"
+	case PermissionTypeDeployments:
+		permName = "deployments"
+	case PermissionTypePackages:
+		permName = "contents"
+	case PermissionTypeActions:
+		permName = "actions"
+	default:
+		permName = ""
+	}
+	return &permName
+}
+
+func createFinding(loct *checker.PermissionLocation, text string, loc *finding.Location) (*finding.Finding, error) {
+	probe := stepsNoWriteID
+	if loct == nil || *loct == checker.PermissionLocationTop {
+		probe = topNoWriteID
+	}
+	content, err := probes.ReadFile(probe + ".yml")
+	if err != nil {
+		return nil, fmt.Errorf("%w", err)
+	}
```

Comment:
add some context to the wrap. "%w" is circumventing the linter

Reply:
Not sure I understood correctly, but added a comment here: https://github.com/ossf/scorecard/pull/3693/commits/e8762564509bd47b452864780d1bb496c0d74262

Reply:
I was talking about the "Wrapping errors with %w" section of this blogpost.
https://go.dev/blog/go1.13-errors#:~:text=Wrapping%20errors%20with%20%25w

The linter's purpose is to add additional information to an error, but `fmt.Errorf("%w", err)` isn't adding anything. Instead, I meant something like:
```go
if err != nil {
	return nil, fmt.Errorf("reading probe file: %w", err)
}
```

or include the probe:
```go
if err != nil {
	return nil, fmt.Errorf("reading %v.yml: %w", probe, err)
}
```

Reply:
Done in https://github.com/ossf/scorecard/pull/3693/commits/ee44701c50df49b8775c9c8089fe47ba2b10c1ed